src/libraries/System.Private.CoreLib/generators/EventSourceGenerator.Emitter.cs

```diff
@@ -62,12 +64,32 @@ partial class {ec.ClassName}
                     _builder.AppendLine($@"
 }}");
                 }
+                
             }
 
-            private void GenerateConstructor(EventSourceClass ec)
+            private void GenerateConstructor(EventSourceClass ec, ITypeSymbol stringTypeSymbol)
             {
                 _builder.AppendLine($@"
-        private {ec.ClassName}() : base(new Guid({ec.Guid.ToString("x").Replace("{", "").Replace("}", "")}), ""{ec.SourceName}"") {{ }}");
+        private {ec.ClassName}() : base(new Guid({ec.Guid.ToString("x").Replace("{", "").Replace("}", "")}), ""{ec.SourceName}"") {{");
+                EventDataBuilder.BuildEventDescriptor(_builder, ec.Events);
+                _builder.AppendLine("        m_EventMetadataInitializer = () => new byte[] {");
```

Comment:
Could do this as a `virtual ReadOnlySpan<byte>` like is done with `ProviderMetadata`?

https://github.com/dotnet/runtime/blob/350ee45ba750efad16b49c920092340e01eaa722/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs#L39

I think most usages would be happy (with some tweaks) taking ROS instead?

Reply:
I'll have to check but I believe so. I'll look into changing m_manifestBytes to ReadOnlySpan.